### PR TITLE
Allow creation of issue comments in cml

### LINKED ISSUE #1180: issue comments
CML version of `gh issue comment`?

Use cases:

- open an issue on failing job (not just a `cml check` or commit-`comment`) & post a report to it
- (ab)use an issue as an experiment dashboard (`main` cron jobs post reports to a hardcoded issue)

## See also

- https://github.com/iterative/cml-stable-diffusion/blob/0416810a916cc29bbfeb9b0124bd82dff1c2d84b/.github/workflows/model.yml
- https://github.com/iterative/status.cml.dev/pull/3
- https://github.com/iterative/status.cml.dev/pull/4

### PR DESCRIPTION
**Pull request by tasdomas**
Closes #1180.

### CODE REVIEW

File: bin/cml/comment/create.js [hunk 1]
```diff
@@ -16,6 +16,12 @@ exports.options = kebabcaseKeys({
     description:
       'Post to an existing PR/MR associated with the specified commit'
   },
+  issue: {
+    type: 'string',
```
Comment (0x2b3bfa0) [inline bin/cml/comment/create.js:20]: Should it take a number instead?

```suggestion
    type: 'number',
```

File: bin/cml/comment/create.js [hunk 2]
```diff
@@ -16,6 +16,12 @@ exports.options = kebabcaseKeys({
...
+    description: 'Issue id to post comment to.',
+    default: '',
```
Comment (0x2b3bfa0) [inline bin/cml/comment/create.js:22]: Not needed? 🤔

File: bin/cml/comment/create.js [hunk 3]
```diff
@@ -16,6 +16,11 @@ exports.options = kebabcaseKeys({
     description:
       'Post to an existing PR/MR associated with the specified commit'
   },
+  issue: {
+    type: 'number',
+    description: 'Issue id to post comment to.',
```
Comment (casperdcl) [inline bin/cml/comment/create.js:21]:
```suggestion
    description: 'Post to the given issue number',
```

File: bin/cml/comment/create.js [hunk 4]
```diff
@@ -16,6 +16,11 @@ exports.options = kebabcaseKeys({
     description:
       'Post to an existing PR/MR associated with the specified commit'
```
Comment (casperdcl) [inline bin/cml/comment/create.js:17]:
```suggestion
      'Post to an existing PR/MR associated with the specified commit',
    conflicts: ['issue']
```

File: bin/cml/comment/create.js [hunk 5]
```diff
@@ -16,6 +16,11 @@ exports.options = kebabcaseKeys({
...
   },
+  issue: {
+    type: 'number',
+    description: 'Issue id to post comment to.',
+    conflicts: ['pr']
```
Comment (casperdcl) [inline bin/cml/comment/create.js:22]:
```suggestion
    conflicts: ['pr', 'commitSha']
```

**Review by casperdcl — CHANGES_REQUESTED** [review]

**Review by dacbd — CHANGES_REQUESTED** [review]
```
  issue: {
    type: 'number',
    description: 'Post to the given issue number',
    conflicts: ['pr', 'commitSha']
  },
```

conflicts with `commitSha` default value.
https://github.com/iterative/cml-playground/actions/runs/3187003743/jobs/5198152076

**Comment by tasdomas** [comment]
> ```
>   issue: {
>     type: 'number',
>     description: 'Post to the given issue number',
>     conflicts: ['pr', 'commitSha']
>   },
> ```
> 
> conflicts with `commitSha` default value. https://github.com/iterative/cml-playground/actions/runs/3187003743/jobs/5198152076

Updated - thanks!

File: src/drivers/github.js [hunk 6]
```diff
@@ -540,6 +540,56 @@ class Github {
...
+    const { id, report } = opts;
+    const { owner, repo } = ownerRepo({ uri: this.repo });
+    const { issues } = octokit(this.token, this.repo);
+
+    const {
+      data: { html_url: htmlUrl }
+    } = await issues.updateComment({
+      owner,
+      repo,
+      report,
```
Comment (dacbd) [inline src/drivers/github.js:570]:
```suggestion
      body: report,
```

File: src/drivers/github.js [hunk 7]
```diff
@@ -540,6 +540,56 @@ class Github {
...
+    const { issueId, report } = opts;
+    const { owner, repo } = ownerRepo({ uri: this.repo });
+    const { issues } = octokit(this.token, this.repo);
+
+    const {
+      data: { html_url: htmlUrl }
+    } = await issues.createComment({
+      owner,
+      repo,
+      report,
```
Comment (dacbd) [inline src/drivers/github.js:553]:
```suggestion
      body: report,
```

**Review by dacbd — CHANGES_REQUESTED** [review]

File: src/cml.js [hunk 8]
```diff
@@ -163,6 +163,7 @@ class CML {
     const triggerSha = await this.triggerSha();
     const {
       commitSha: inCommitSha = triggerSha,
+      issue: issueId,
```
Comment (casperdcl) [inline src/cml.js:166]: bugging me as potential source of lots of future dev headache.

It's technically `issueNumber`, not `issueId`, right?

`gh issue view $NUMBER --json id --jq.id` != `$NUMBER`

Comment (tasdomas) [reply src/cml.js:166]: github is not the only forge we support and on bitbucket and gitlab issue id is used.

Comment (dacbd) [reply src/cml.js:166]: the flag is still just `--issue` I don't feel this variable name that's not exposed to the user is a concern

**Review by dacbd — APPROVED** [review]

**Comment by tasdomas** [comment]
@casperdcl I was planning to submit smaller PRs into the feature branch and then propose landing it in master once its done.

**Comment by casperdcl** [comment]
May as well open a draft PR as well? Makes it easier to track (at least for me)

**Pull request merged**